src/debugger/ios/iOSPlatform.ts

```diff
@@ -45,8 +50,11 @@ export class IOSPlatform implements IAppPlatform {
             // React native supports running on the iOS simulator from the command line
             let runArguments: string[] = [];
             if (this.simulatorTarget.toLowerCase() !== IOSPlatform.simulatorString) {
-                runArguments.push("--simulator");
-                runArguments.push(this.simulatorTarget);
+                runArguments.push("--simulator", this.simulatorTarget);
+            }
+
+            if (path.join(this.projectPath, IOSPlatform.DEFAULT_IOS_PROJECT_RELATIVE_PATH) !== this.iosProjectPath) {

```

Comment:
I think it'd be simpler to remove the if, and just pass --project-path always to run-ios